drivers/lora/sx1276.c

```diff
@@ -112,51 +109,61 @@ void BoardCriticalSectionEnd(uint32_t *mask)
 	irq_unlock(*mask);
 }
 
-uint32_t RtcGetTimerElapsedTime(void)
+u32_t RtcGetTimerValue(void)
 {
-	u32_t ticks;
-	int err;
-
-	err = counter_get_value(dev_data.counter, &ticks);
-	if (err) {
-		LOG_ERR("Failed to read counter value (err %d)", err);
-		return 0;
-	}
+	return k_ms_to_ticks_ceil32(k_uptime_get_32());
+}
 
-	return ticks;
+u32_t RtcGetTimerElapsedTime(void)
+{
+	return (k_ms_to_ticks_ceil32(k_uptime_get_32()) - saved_time);
 }
 
 u32_t RtcGetMinimumTimeout(void)
 {
-	/* TODO: Get this value from counter driver */
-	return 3;
+	return 1;
 }
 
-void RtcSetAlarm(uint32_t timeout)
+void RtcStopAlarm(void)
 {
-	struct counter_alarm_cfg alarm_cfg;
+	k_timer_stop(&dev_data.timer);
+}
 
-	alarm_cfg.flags = 0;
-	alarm_cfg.ticks = timeout;
+static void timer_callback(struct k_timer *_timer)
+{
+	ARG_UNUSED(_timer);
 
-	counter_set_channel_alarm(dev_data.counter, 0, &alarm_cfg);
+	TimerIrqHandler();
 }
 
-uint32_t RtcSetTimerContext(void)
+void RtcSetAlarm(u32_t timeout)
 {
-	return 0;
+	k_timer_start(&dev_data.timer, k_ticks_to_ms_floor32(timeout),
```

Comment:
New `k_timer_start()` gets `k_timeout_t` arguments instead of `s32_t`. And those are not miliseconds anymore if you disable `CONFIG_LEGACY_TIMEOUT_API` (once enough code will be converted).

Reply:
Will convert the whole driver to new timeout API

Reply:
Since this involves using the rtc, you may want to have a look at the discussion in #23526

Reply:
@sslupsky We are getting rid of RTC in LoRa. This is mainly due to the limitation of some of the RTC drivers which provide 1s granularity.